python/ray/tune/execution/ray_trial_executor.py

```diff
@@ -235,43 +234,50 @@ def __init__(
         else:
             self._trial_cleanup = None
 
+        # For printing used resources
         self._resource_updater = _ResourceUpdater(refresh_period)
 
-        self._has_cleaned_up_pgs = False
+        # Actor re-use
         self._reuse_actors = reuse_actors
-        # The maxlen will be updated when `setup(max_pending_trials)` is called
-        self._cached_actor_pg = deque(maxlen=1)
-        self._pg_manager = _PlacementGroupManager(prefix=_get_tune_pg_prefix())
+        self._max_staged_actors = 1
+        self._cached_resources_to_actor = defaultdict(list)
+        self._last_cached_actor_cleanup = float("-inf")
+        # Protect these actors from the first cleanup
```

Comment:
What is the problem it's trying to solve?

Reply:
At the moment of caching, we haven't created a new trial, yet. Thus we don't know if we actually need to cache this actor or not. We only know this on the next iteration of `step()`.

I can add a more thorough comment here